### PR TITLE
Run regresscheck on PG13

### PR DESCRIPTION
**Pull request by svenklemm**

### CODE REVIEW

File: test/expected/partitionwise-13.out [hunk 1]
```diff
@@ -0,0 +1,884 @@
...
+               Presorted Key: (time_func(_hyper_3_7_chunk."time"))
+               ->  Result
+                     Output: (time_func(_hyper_3_7_chunk."time")), _hyper_3_7_chunk.device, _hyper_3_7_chunk.temp
+                     ->  Merge Append
+                           Sort Key: (time_func(_hyper_3_7_chunk."time"))
+                           ->  Index Scan Backward using _hyper_3_7_chunk_hyper_timepart_expr_idx on _timescaledb_internal._hyper_3_7_chunk
+                                 Output: _hyper_3_7_chunk."time", _hyper_3_7_chunk.device, _hyper_3_7_chunk.temp, time_func(_hyper_3_7_chunk."time")
+                           ->  Index Scan Backward using _hyper_3_8_chunk_hyper_timepart_expr_idx on _timescaledb_internal._hyper_3_8_chunk
+                                 Output: _hyper_3_8_chunk."time", _hyper_3_8_chunk.device, _hyper_3_8_chunk.temp, time_func(_hyper_3_8_chunk."time")
+(17 rows)
```
Comment (mkindahl) [inline test/expected/partitionwise-13.out:788]: This is different compared to PG12. Have you checked that it is OK?

File: test/expected/partitionwise-13.out [hunk 2]
```diff
@@ -0,0 +1,884 @@
...
+                     Sort Key: (time_func(_hyper_3_7_chunk."time")), _hyper_3_7_chunk.device
+                     Presorted Key: (time_func(_hyper_3_7_chunk."time"))
+                     ->  Index Scan Backward using _hyper_3_7_chunk_hyper_timepart_expr_idx on _timescaledb_internal._hyper_3_7_chunk
+                           Output: time_func(_hyper_3_7_chunk."time"), _hyper_3_7_chunk.device, _hyper_3_7_chunk.temp
+         ->  GroupAggregate
+               Output: (time_func(_hyper_3_8_chunk."time")), _hyper_3_8_chunk.device, avg(_hyper_3_8_chunk.temp)
+               Group Key: (time_func(_hyper_3_8_chunk."time")), _hyper_3_8_chunk.device
+               ->  Incremental Sort
+                     Output: (time_func(_hyper_3_8_chunk."time")), _hyper_3_8_chunk.device, _hyper_3_8_chunk.temp
+                     Sort Key: (time_func(_hyper_3_8_chunk."time")), _hyper_3_8_chunk.device
```
Comment (mkindahl) [inline test/expected/partitionwise-13.out:846]: This is different from PG12. Have you checked it?

File: test/expected/partitionwise-13.out [hunk 3]
```diff
@@ -0,0 +1,884 @@
...
+                           Output: time_func(_hyper_3_7_chunk."time"), _timescaledb_internal.get_partition_hash(_hyper_3_7_chunk.device), _hyper_3_7_chunk.temp
+         ->  GroupAggregate
+               Output: (time_func(_hyper_3_8_chunk."time")), (_timescaledb_internal.get_partition_hash(_hyper_3_8_chunk.device)), avg(_hyper_3_8_chunk.temp)
+               Group Key: (time_func(_hyper_3_8_chunk."time")), (_timescaledb_internal.get_partition_hash(_hyper_3_8_chunk.device))
+               ->  Incremental Sort
+                     Output: (time_func(_hyper_3_8_chunk."time")), (_timescaledb_internal.get_partition_hash(_hyper_3_8_chunk.device)), _hyper_3_8_chunk.temp
+                     Sort Key: (time_func(_hyper_3_8_chunk."time")), (_timescaledb_internal.get_partition_hash(_hyper_3_8_chunk.device))
+                     Presorted Key: (time_func(_hyper_3_8_chunk."time"))
+                     ->  Index Scan Backward using _hyper_3_8_chunk_hyper_timepart_expr_idx on _timescaledb_internal._hyper_3_8_chunk
+                           Output: time_func(_hyper_3_8_chunk."time"), _timescaledb_internal.get_partition_hash(_hyper_3_8_chunk.device), _hyper_3_8_chunk.temp
```
Comment (mkindahl) [inline test/expected/partitionwise-13.out:882]: This is different from PG12. Have you checked it?

File: test/sql/include/join_query.sql [hunk 4]
```diff
@@ -2160,6 +2160,7 @@ rollback to settings;
 savepoint settings;
 set local max_parallel_workers_per_gather = 0;
 set local work_mem = '128kB';
+set enable_mergejoin to false;
```
Comment (mkindahl) [inline test/sql/include/join_query.sql:2163]: You're disabling mergejoin here, but do we have similar merge joins in other tests?

**Review by mkindahl — COMMENTED** [review]
I compared the output files for PG12 and PG13 to see that there were only benign changes. Here's a summary:

- These tests are identical in PG12 and PG13: `custom_type`, `multi_transaction_index`, `parallel`, `partition`, `chunk_adaptive`, `plan_hashagg`, `query`, `sql_query`
- These tests have new names in the plans, but otherwise match: `delete`, `plan_expand_hypertable`, `plan_hypertable_inline`, `update`
- Test `partitionwise` have some changes in plans, please check below and verify that they are OK.
- These tests have changes in error messages: `constraint`, `ddl`, `insert`

File: test/sql/include/join_query.sql (hunk 4 above)
Comment (svenklemm) [reply test/sql/include/join_query.sql]: this test requires hashjoin and doesnt apply to mergejoin

File: test/expected/partitionwise-13.out (hunk 1 above)
Comment (svenklemm) [reply test/expected/partitionwise-13.out:788]: yes

File: test/expected/partitionwise-13.out (hunk 2 above)
Comment (svenklemm) [reply test/expected/partitionwise-13.out:846]: yes

File: test/expected/partitionwise-13.out (hunk 3 above)
Comment (svenklemm) [reply test/expected/partitionwise-13.out:882]: yes

**Review by mkindahl — APPROVED** [review]

File: test/expected/partitionwise-13.out [hunk 5]
```diff
@@ -0,0 +1,884 @@
...
+SELECT time_func(time), device, avg(temp)
+FROM hyper_timepart
+GROUP BY 1, 2
+ORDER BY 1, 2
+LIMIT 10;
+                                                                     QUERY PLAN                                                                      
+-----------------------------------------------------------------------------------------------------------------------------------------------------
+ Limit
+   Output: (time_func(_hyper_3_7_chunk."time")), _hyper_3_7_chunk.device, (avg(_hyper_3_7_chunk.temp))
+   ->  GroupAggregate
```
Comment (gayyappan) [inline test/expected/partitionwise-13.out:773]: I see some of the HashAggregate became GroupAggregates (for thsi file) in PG13. Do we need to set enable_hashagg = false here to ensure that plans will not change?

**Review by gayyappan — APPROVED** [review]
Looks good. Have a question about 1 of the tests.

File: test/expected/partitionwise-13.out (hunk 5 above)
Comment (svenklemm) [reply test/expected/partitionwise-13.out:773]: i think the groupagg plans are preferable and i think we dont care what type of aggregation postgres uses here and the test is already pg-version specific so it should be stable

**Pull request merged**